src/sudoers/mod.rs

```diff
@@ -193,15 +193,46 @@ impl Sudoers {
                 }
             });
 
-        if let Some(Tag { authenticate, .. }) = flags.as_mut() {
+        if let Some(tag) = flags.as_mut() {
             if skip_passwd {
-                *authenticate = Authenticate::Nopasswd;
+                tag.authenticate = Authenticate::Nopasswd;
             }
+
+            Authorization::Allowed(self.settings.to_auth(tag), ())
+        } else {
+            Authorization::Forbidden
         }
+    }
 
-        Judgement {
-            flags,
-            settings: self.settings.clone(), // this is wasteful, but in the future this will not be a simple clone and it avoids a lifetime
+    pub fn check_validate_permission<User: UnixUser + PartialEq<User>>(
+        &self,
+        invoking_user: &User,
+        hostname: &system::Hostname,
+    ) -> Authorization {
+        // exception: if user is root, NOPASSWD is implied
+        let skip_passwd = invoking_user.is_root();
+
+        let mut flags = self
+            .matching_user_specs(invoking_user, hostname)
+            .flatten()
+            .fold(None::<Tag>, |outcome, (_, (tag, _))| {
```

Comment:
I don't like this `.fold()`. Every fold is simply a fancypants loop, and I think the logic isn't very clearly expressed as well (it returns the first tag if all of them have NOPASSWD, but the last one that has PASSWD if there are some?).

But since this code is similar to the `check_list_permission` one I'll wave it through here and make a follow-up PR where I'll do some beautification.